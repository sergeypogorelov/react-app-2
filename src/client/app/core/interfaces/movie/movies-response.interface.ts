import { DataResponse } from '../common/data-response.interface';
import { Movie } from './movie.interface';

export interface MoviesResponse extends DataResponse<Movie> {}
