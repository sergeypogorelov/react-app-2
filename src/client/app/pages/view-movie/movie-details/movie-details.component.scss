@import '../../../../styles/variables';

.movie-details {
  &-info {
    padding: 30px 120px 0 60px;
  }
  &-poster {
    width: 330px;
  }
  &-rating {
    display: flex;
    justify-content: center;
    align-items: center;
    width: 70px;
    height: 70px;
    font-size: 26px;
    color: rgba($color: #a1e66f, $alpha: 0.8);
    border: 1px solid rgba($color: $c-default, $alpha: 0.8);
    border-radius: 50%;
  }
  &-subtitle {
    font-family: $f-medium;
    font-size: 16px;
    color: rgba($color: $c-default, $alpha: 0.7);
  }
  &-additional {
    .movie-additional-info {
      margin-right: 30px;
    }
    display: flex;
    margin-top: 30px;
    margin-bottom: 25px;
  }
  &-description {
    font-size: 17px;
    line-height: 22px;
    color: rgba($color: $c-default, $alpha: 0.8);
  }
  .header-title {
    margin-right: 20px;
  }
}
