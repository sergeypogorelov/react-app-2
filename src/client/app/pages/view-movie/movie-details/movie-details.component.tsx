import './movie-details.component.scss';

import React, { FunctionComponent } from 'react';

import { imageKillBill } from '../../../core/constants/assets-urls';

import { MovieAdditionalInfoProps } from './movie-additional-info/movie-additional-info-props.interface';

import { HeaderTitle } from '../../../shared/header/header-title/header-title.component';
import { MovieAdditionalInfo } from './movie-additional-info/movie-additional-info.component';

export const MovieDetails: FunctionComponent<{}> = () => {
  const items: MovieAdditionalInfoProps[] = [
    {
      value: '1994',
      unit: 'year',
    },
    {
      value: '154',
      unit: 'min',
    },
  ];

  return (
    <div className="movie-details d-flex">
      <img className="movie-details-poster" src={imageKillBill} />
      <div className="movie-details-info">
        <div className="movie-details-title d-flex align-items-center">
          <HeaderTitle>Kill Bill: Vol 2</HeaderTitle>
          <div className="movie-details-rating">4.3</div>
        </div>
        <div className="movie-details-subtitle">Oscar winning Movie</div>
        <div className="movie-details-additional">
          {items.map((item, index) => (
            <MovieAdditionalInfo key={index} {...item} />
          ))}
        </div>
        <div className="movie-details-description">
          Pulp Fiction is a 1994 American crime film written and directed by
          Quentin Tarantino; it is based on a story by Tarantino and Roger
          Avary.[4] Starring John Travolta, Samuel L. Jackson, Bruce Willis, Tim
          Roth, Ving Rhames, and Uma Thurman, it tells several stories of
          criminal Los Angeles. The film's title refers to the pulp magazines
          and hardboiled crime novels popular during the mid-20th century, known
          for their graphic violence and punchy dialogue.
        </div>
      </div>
    </div>
  );
};
