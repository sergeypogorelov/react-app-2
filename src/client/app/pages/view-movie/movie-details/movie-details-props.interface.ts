import { Movie } from '../../../core/interfaces/movie/movie.interface';

export interface MovieDetailsProps {
  movie: Movie;
}
