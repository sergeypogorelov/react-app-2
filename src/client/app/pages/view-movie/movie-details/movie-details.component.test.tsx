import React from 'react';
import renderer from 'react-test-renderer';

import { MovieDetails } from './movie-details.component';
import { Movie } from '../../../core/interfaces/movie/movie.interface';

test('MovieDetails component should render correctly', () => {
  const movie: Movie = {
    id: 1,
    title: 'title',
    tagline: 'tagline',
    vote_average: 10,
    vote_count: 100,
    release_date: '1993-12-03',
    poster_path: 'poster path',
    overview: 'overview',
    budget: 1000000,
    revenue: 1000,
    runtime: 150,
    genres: ['genre'],
  };
  const tree = renderer.create(<MovieDetails movie={movie} />).toJSON();
  expect(tree).toMatchSnapshot();
});
