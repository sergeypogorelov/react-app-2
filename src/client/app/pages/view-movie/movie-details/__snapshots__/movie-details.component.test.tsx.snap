// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails component should render correctly 1`] = `
<div
  className="movie-details d-flex"
>
  <img
    className="movie-details-poster"
    onError={[Function]}
    src="poster path"
  />
  <div
    className="movie-details-info"
  >
    <div
      className="movie-details-title d-flex align-items-center"
    >
      <h2
        className="header-title"
      >
        title
      </h2>
      <div
        className="movie-details-rating"
      >
        10
      </div>
    </div>
    <div
      className="movie-details-subtitle"
    >
      tagline
    </div>
    <div
      className="movie-details-additional"
    >
      <div
        className="movie-additional-info"
      >
        <span
          className="movie-additional-info-value"
        >
          1993
        </span>
        <span
          className="movie-additional-info-unit"
        >
          year
        </span>
      </div>
      <div
        className="movie-additional-info"
      >
        <span
          className="movie-additional-info-value"
        >
          150
        </span>
        <span
          className="movie-additional-info-unit"
        >
          min
        </span>
      </div>
    </div>
    <div
      className="movie-details-description"
    >
      overview
    </div>
  </div>
</div>
`;
